internal/injector/aspect/join/function.go

```diff
@@ -350,6 +361,133 @@ func (s *functionBody) Hash(h *fingerprint.Hasher) error {
 	return h.Named("function-body", s.Function)
 }
 
+// resultImplements matches functions where at least one return value's type
+// implements the specified interface.
+type resultImplements struct {
+	InterfaceName string
+}
+
+// ResultImplements creates a FunctionOption that matches functions where at least one
+// return value implements the named interface.
+func ResultImplements(interfaceName string) FunctionOption {
+	return &resultImplements{InterfaceName: interfaceName}
+}
+
+func (fo *resultImplements) impliesImported() []string {
+	pkgPath, _ := typed.SplitPackageAndName(fo.InterfaceName)
+	if pkgPath != "" {
+		return []string{pkgPath}
+	}
+	return nil
+}
```

Comment:
I don't think returning some type that implements an interface implies having imported the package where the interface is defined...